kafka/consumer.go

```diff
@@ -482,12 +482,20 @@ func (c *Consumer) GetMetadata(topic *string, allTopics bool, timeoutMs int) (*M
 	return getMetadata(c, topic, allTopics, timeoutMs)
 }
 
-// QueryWatermarkOffsets returns the broker's low and high offsets for the given topic
+// QueryWatermarkOffsets queries the broker for the low and high offsets for the given topic
 // and partition.
 func (c *Consumer) QueryWatermarkOffsets(topic string, partition int32, timeoutMs int) (low, high int64, err error) {
 	return queryWatermarkOffsets(c, topic, partition, timeoutMs)
 }
 
+// GetWatermarkOffsets returns the cached low and high offsets for the given topic
+// and partition.  The high offset is populated on every fetch response or via calling QueryWatermarkOffsets.
+// The low offset is populated only when calling QueryWatermarkOffsets.  0 will be returned if
```

Comment:
The low offset is populated every statistics.interval.ms or when calling QueryWatermarkOffsets.

Reply:
Actually, going through this now, it looks like _nothing_ populates the low offset (`rktp->rktp_lo_offset`) that is returned by `get_watermark_offsets`

https://github.com/edenhill/librdkafka/search?q=rktp_lo_offset&unscoped_q=rktp_lo_offset

Reply:
It is only updated when statistics is enabled.

Reply:
Changed the wording in my comment above.